[IOS_FAMILY] Drop configurations for 2021-aligned OSes

litherum opened the pull request:
#### e499616cb404edad37b4bfab3c61eec2b9886ee6
<pre>
[IOS_FAMILY] Drop configurations for 2021-aligned OSes
<a href="https://bugs.webkit.org/show_bug.cgi?id=258423">https://bugs.webkit.org/show_bug.cgi?id=258423</a>
rdar://111189039

Reviewed by Alexey Proskuryakov.

All supported IOS_FAMILY OSes are later than the 2021-aligned ones.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:

Canonical link: <a href="https://commits.webkit.org/265437@main">https://commits.webkit.org/265437@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/543691e776ce8a764178d691e36b988ee13e9fb7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10897 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11110 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11423 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12544 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10421 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13487 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11091 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13337 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11056 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/11959 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9180 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12948 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9252 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9841 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17075 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9223 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10323 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9993 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/13230 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10333 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10440 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8529 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11006 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9612 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3008 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2614 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13882 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11315 "Built successfully") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/10300 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2773 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -151,7 +151,7 @@
 #endif
 
-#if !defined(ENABLE_APPLE_PAY_AMS_UI) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 150400) || PLATFORM(VISION))
+#if !defined(ENABLE_APPLE_PAY_AMS_UI) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) || PLATFORM(IOS) || PLATFORM(VISION))
```
I think macOS 12 is 2021 too

litherum replied:
12.3 is a point release of Monterey, which we still support.

aproskuryakov replied:
As discussed in Slack, no, we don't support SDKs older than 12.3.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -319,15 +319,9 @@
-    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED >= 80000) \
-    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MIN_REQUIRED >= 150000) \
-    || PLATFORM(VISION)
+#if !defined(ENABLE_GPU_PROCESS_BY_DEFAULT) && PLATFORM(COCOA)
```
`PLATFORM(COCOA)` isn't needed in this file.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -789,13 +783,13 @@
 #endif
 
-#if !defined(ENABLE_APP_PRIVACY_REPORT) && PLATFORM(IOS_FAMILY) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 150000
+#if !defined(ENABLE_APP_PRIVACY_REPORT) && PLATFORM(IOS_FAMILY)
```
This one is weird, I don't know where `PLATFORM(IOS_FAMILY) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 150000` is true. I think that this may exclude watchOS and tvOS, but maybe not visionOS?

litherum replied:
It seems likely that whoever wrote this just made a mistake, and meant to enable this feature on the 2021-aligned OSes.

litherum replied:
I think you're right, that, to make sure these patches don't have behavior changes, I should exclude any OSes which don't set `__IPHONE_OS_VERSION_MIN_REQUIRED` which I believe is just watchOS and tvOS.

aproskuryakov replied:
I think that they set it to some frozen value that's smaller than 150000. And visionOS probably sets it to 170000, but I'm just guessing. As I said, this is a weird one.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -789,13 +783,13 @@
 #endif
 
 #if (!defined(ENABLE_APP_HIGHLIGHTS) \
     && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) \
```
Same, macOS should be part of this PR.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -832,7 +826,7 @@
 #endif
 
-#if (ENABLE(ARKIT_INLINE_PREVIEW_MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) || (ENABLE(ARKIT_INLINE_PREVIEW_IOS) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 150400)
+#if (ENABLE(ARKIT_INLINE_PREVIEW_MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) || ENABLE(ARKIT_INLINE_PREVIEW_IOS)
```
Same.

litherum replied:
Right, same response about how 12.3 is a point release of a supported OS

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -367,7 +367,7 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) \
```
Same.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -692,7 +692,7 @@
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 150400) || PLATFORM(VISION)
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) || PLATFORM(IOS) || PLATFORM(VISION)
```
Same

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -962,9 +962,7 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) \
```
Same. Collapses to COCOA I think.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1210,13 +1206,11 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 120300) \
```
Same.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1262,7 +1256,7 @@
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000) || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED < 150000)
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000) || PLATFORM(MACCATALYST)
```
This change looks incorrect.

litherum replied:
Yep, thanks for catching!

aproskuryakov reviewed (approved):
I think there is one change at the bottom that's incorrect.

litherum committed:
[IOS_FAMILY] Drop configurations for 2021-aligned OSes
https://bugs.webkit.org/show_bug.cgi?id=258423
rdar://111189039

Reviewed by Alexey Proskuryakov.

All supported IOS_FAMILY OSes are later than the 2021-aligned ones.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:

Canonical link: https://commits.webkit.org/265437@main
